torch/distributed/_shard/sharding_spec/api.py

```diff
@@ -110,3 +110,62 @@ def __post_init__(self):
             rank = len(shard.shard_offsets)
 
         validate_non_overlapping_shards_metadata(self.shards)
+
+
+def _infer_sharding_spec_from_shards_metadata(shards_metadata):
+    """
+    Infer the sharding spec from the metadata of each shard of a ShardedTensor.
+    If the tensor is sharded only on one dimension, we then assume it's chunk sharded.
+    Otherwise, we assume it's enum sharded.
+
+    Args:
+        shards_metadata (List[ShardMetadata]): List of Metadata of local shards.
+
+    Returns:
+        A :class:`torch.distributed._shard.sharding_spec.ShardingSpec` object of sharding
+            spec for one sharded tensor.
+    """
+    placements = []
+    chunk_sharding_dim = None
+    chunk_offset_list = []
+    shard_size_list = []
+    # collect local shard metadatas from the global sharded_tensor_metadata
+    for shard_metadata in shards_metadata:  # type: ignore[attr-defined]
+        placements.append(shard_metadata.placement)
+        local_offsets = shard_metadata.shard_offsets
+        chunk_offset_list.append(sum(local_offsets))
```

Comment:
why we need to sum the local_offsets list?

Reply:
Oh, because for here, we don't know the sharding col, so we just directly sum it. 

The assumption here is that, for chunk sharding case, the offset will always be [0, 0, ..., N, ..., 0]. So if we sum it, we  will get N regardless its sharding col.